File: algorithms/ar-cochprb/Javascript/coin_change.js
function change(coins, i, amount) {
  if (amount === 0) {
    return 1;
  }
  if (amount < 0) {
    return 0;
  }
  if (i <= 0) {
    return 0;
  }

  return (
    change(coins, i - 1, coins[i - 1]) + change(coins, i, amount - coins[i - 1])
  );
}
let coinArr = [1, 2, 3];
console.log(change(coinArr, coinArr.length, 4));

File: algorithms/ar-cochprb/python2/coinchange.py
def countcombo(coins, n, sum):#using DP approach, google for more
    if (sum == 0):
        return 1
    if (sum < 0):
        return 0
    if (n <= 0):
        return 0
    return countcombo(coins, n - 1, sum) + countcombo(coins, n, sum-coins[n-1])

if __name__=="__main__":
    coin_arr = [int(i) for i in raw_input("Enter unique(non repeating ) coins seperated by space :").split()]
    #removing duplicates if any
    coin_arr = list(set(coin_arr))
    s = int(raw_input("\nEnter needed sum : "))
    print"\n\nAll possible coin change for enter sum ",s," is ",countcombo(coin_arr,len(coin_arr),s)

